apisix/ssl.lua

```diff
@@ -252,9 +253,14 @@ function _M.check_ssl_conf(in_dp, conf)
         end
     end
 
-    local ok, err = validate(conf.cert, conf.key)
-    if not ok then
-        return nil, err
+    -- if the certificate or key uses a secret reference, we only verify it when using it
+    if not secret.check_secret_uri(conf.cert) or
+        not secret.check_secret_uri(conf.key) then
+
+        local ok, err = validate(conf.cert, conf.key)
```

Comment:
If the cert is using secret_uri and the key doesn't, the validate will failed.
We have two choices:
1. the key and cert should support secret_uri both
2. validate cert and key seperately

Could we add test cases about this?

Reply:
The check_ssl_conf function is called in two places:
1. cp plane write configuration
2. The dp plane reads the configuration from the configuration center
When a cert or key is referenced by a secret, it is not verified for the following reasons:
1. Do not do too many dynamic checks on the cp plane, which can simplify the network topology
2. The dp plane will verify when using the cert and key
3. 
Therefore, it is only verified when neither the cert nor the key is a secret reference

Reply:
OK, when the key or cert is  a secret reference, validate them later